src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Attr.java

```diff
@@ -1658,8 +1657,15 @@ private void handleSwitch(JCTree switchTree,
             boolean enumSwitch = (seltype.tsym.flags() & Flags.ENUM) != 0;
```

Comment:
This is getting convoluted enough that an enum on the AST (e.g. SwitchKind) might be helpful to save some of these classification properties - which I imagine we have to redo at some point later (e.g. in Lower/TransPattern). I'm ok with fixing in a followup issue.

Reply:
Thanks Maurizio. Yes, some of the logic is partly repeated elsewhere, but I need to investigate how to improve that. I've filled:
https://bugs.openjdk.java.net/browse/JDK-8267929